### PR TITLE
+ README Changes

### PR DESCRIPTION
**Pull request by snavinch**

### CODE REVIEW

File: Client/ApiClient.cs [hunk 1]
```diff
@@ -343,6 +355,18 @@ public ApiClient(String basePath = "https://apitest.cybersource.com")
                     RestClient.Proxy = Configuration.Proxy;
                 }            
 
+                // Adding Client Cert
+                if(Equals(bool.Parse(Configuration.MerchantConfigDictionaryObj["enableClientCert"]), true))
```
Comment (gnongsie) [inline Client/ApiClient.cs:359]: This condition will not always work. 

- What happens if I have set `enableClientCert` to `true` but I am trying to use Payments API ? 
- What happens if I have set `enableClientCert` to `true` but I am trying to hit non-MutualAuth environment?

File: README.md [hunk 2]
```diff
@@ -31,6 +31,8 @@ Additionally, you can find details and examples of how our API is structured in
 
 The API Reference Guide provides examples of what information is needed for a particular request and how that information would be formatted. Using those examples, you can easily determine what methods would be necessary to include that information in a request using this SDK.
 
+## MetaKey Support
```
Comment (gnongsie) [inline README.md:34]: Replace this content with what is written in Client SDK for Java.

Also, content for OAuth is missing.

File: generator/cybersource-csharp-template/ApiClient.mustache [hunk 3]
```diff
@@ -228,12 +229,28 @@ namespace {{packageName}}.Client
             {
                 if (postBody.GetType() == typeof(String))
                 {
-                    {{#netStandard}}
-                    request.AddParameter(new Parameter { Value = postBody, Type = ParameterType.RequestBody, ContentType = "application/json" });
-                    {{/netStandard}}
-                    {{^netStandard}}
-                    request.AddParameter("application/json", postBody, ParameterType.RequestBody);
-                    {{/netStandard}}
+					if (contentType == "application/x-www-form-urlencoded")
```
Comment (gnongsie) [inline generator/cybersource-csharp-template/ApiClient.mustache:232]: Indentation needs to be fixed.

File: generator/cybersource-csharp-template/ApiClient.mustache [hunk 4]
```diff
@@ -228,12 +229,28 @@ namespace {{packageName}}.Client
...
+                    {
+                        var bodyParams = JsonConvert.DeserializeObject<Dictionary<string, string>>((string)postBody);
+                        foreach (KeyValuePair<string, string> param in bodyParams)
+                        {
+							{{#netStandard}}
+							request.AddParameter(new Parameter { Value = param.Value, Type = ParameterType.GetOrPost, Name = param.Key });
+							{{/netStandard}}
+							{{^netStandard}}
+							request.AddParameter(param.Key, param.Value, ParameterType.GetOrPost);
+							{{/netStandard}}
```
Comment (gnongsie) [inline generator/cybersource-csharp-template/ApiClient.mustache:242]: Reduce indentation here.

File: generator/cybersource-csharp-template/ApiClient.mustache [hunk 5]
```diff
@@ -228,12 +229,28 @@ namespace {{packageName}}.Client
...
+                        }
+                    }
+                    else
+                    {
+                        {{#netStandard}}
+						request.AddParameter(new Parameter { Value = postBody, Type = ParameterType.RequestBody, ContentType = "application/json" });
+						{{/netStandard}}
+						{{^netStandard}}
+						request.AddParameter("application/json", postBody, ParameterType.RequestBody);
+						{{/netStandard}}
```
Comment (gnongsie) [inline generator/cybersource-csharp-template/ApiClient.mustache:252]: Reduce indentation here.

File: generator/cybersource-csharp-template/ApiClient.mustache [hunk 6]
```diff
@@ -378,6 +395,18 @@ namespace {{packageName}}.Client
                 {
                     RestClient.Proxy = Configuration.Proxy;
                 }            
+				
+				// Adding Client Cert
```
Comment (gnongsie) [inline generator/cybersource-csharp-template/ApiClient.mustache:399]: Indentation needs to be fixed.

File: generator/cybersource-csharp-template/ApiClient.mustache [hunk 7]
```diff
@@ -828,6 +857,11 @@ namespace {{packageName}}.Client
                 if (merchantConfig.IsPostRequest || merchantConfig.IsPutRequest || merchantConfig.IsPatchRequest)
                     authenticationHeaders.Add("Digest", httpSign.Digest);
             }
+			else if (merchantConfig.IsOAuthTokenAuthType)
```
Comment (gnongsie) [inline generator/cybersource-csharp-template/ApiClient.mustache:860]: Indentation needs to be fixed.

**Review by gnongsie — CHANGES_REQUESTED** [review]

File: Client/ApiClient.cs (hunk 1 above)
Comment (snavinch) [reply Client/ApiClient.cs:359]: It'll work. Sending client cert won't create an issue.

File: generator/cybersource-csharp-template/ApiClient.mustache (hunk 3 above)
Comment (snavinch) [reply generator/cybersource-csharp-template/ApiClient.mustache:232]: Done

File: generator/cybersource-csharp-template/ApiClient.mustache (hunk 4 above)
Comment (snavinch) [reply generator/cybersource-csharp-template/ApiClient.mustache:242]: Done

File: generator/cybersource-csharp-template/ApiClient.mustache (hunk 5 above)
Comment (snavinch) [reply generator/cybersource-csharp-template/ApiClient.mustache:252]: Done

File: generator/cybersource-csharp-template/ApiClient.mustache (hunk 6 above)
Comment (snavinch) [reply generator/cybersource-csharp-template/ApiClient.mustache:399]: Done

File: generator/cybersource-csharp-template/ApiClient.mustache (hunk 7 above)
Comment (snavinch) [reply generator/cybersource-csharp-template/ApiClient.mustache:860]: Done

File: README.md (hunk 2 above)
Comment (snavinch) [reply README.md:34]: Done

**Review by gnongsie — APPROVED** [review]

**Pull request merged**